plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityVariant.java

```diff
@@ -0,0 +1,47 @@
+package com.denizenscript.denizen.objects.properties.entity;
+
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.utilities.Utilities;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import org.bukkit.entity.Wolf;
+
+public class EntityVariant extends EntityProperty<ElementTag> {
+
+    // <--[property]
+    // @object EntityTag
+    // @name variant
+    // @input ElementTag
+    // @description
+    // Controls which variant a wolf is.
+    // Valid variants: ASHEN, BLACK, CHESTNUT, PALE, RUSTY, SNOWY, SPOTTED, STRIPED, WOODS
+
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof Wolf;
+    }
+
+    @Override
+    public ElementTag getPropertyValue() {
+        return new ElementTag(as(Wolf.class).getVariant().toString());
```

Comment:
`toString` technically works because the internal `CraftVariant` implementation returns `key.toString()`, but it's probably not great to depend on that (also pretty sure this will return everything with the `minecraft:` namespace?) - probably best to `#getKey` + `Utilities#namespacedKeyToString`.